### PR TITLE
test: 💍 Update test selectors for update host test

### PR DESCRIPTION
**Pull request by calcaide**
# Description

Refactor hosts update test with updated test selectors.

Minor refactor on what field we use to test form error, left detailed comment in code changes.

:tickets: [Jira ticket](https://hashicorp.atlassian.net/browse/ICU-15499)

## How to Test
- Run test suite successfully.

### CODE REVIEW

File: ui/admin/tests/acceptance/host-catalogs/hosts/create-test.js [hunk 1]
```diff
@@ -129,6 +129,8 @@ module('Acceptance | host-catalogs | hosts | create', function (hooks) {
     assert.strictEqual(getHostCount(), count);
   });
 
+  // TODO: the field should change to address and not name. Name is optional for host but address
```
Comment (calcaide) [inline ui/admin/tests/acceptance/host-catalogs/hosts/create-test.js:132]: This will be solve shortly in a separate PR, despite is a small change, I think it does not belong to the specific scope of this PR.

File: ui/admin/tests/acceptance/host-catalogs/hosts/update-test.js [hunk 2]
```diff
@@ -68,32 +70,47 @@ module('Acceptance | host-catalogs | hosts | update', function (hooks) {
...
-    await click('form [type="button"]', 'Activate edit mode');
-    await fillIn('[name="name"]', 'random string');
-    await click('.rose-form-actions [type="submit"]');
-    assert.strictEqual(currentURL(), urls.host);
-    assert.strictEqual(
-      this.server.schema.hosts.all().models[0].name,
-      'random string',
+
+    await click(commonSelectors.EDIT_BTN, 'Activate edit mode');
+    await fillIn(commonSelectors.FIELD_NAME, commonSelectors.FIELD_NAME_VALUE);
```
Comment (calcaide) [inline ui/admin/tests/acceptance/host-catalogs/hosts/update-test.js:77]: Apart from only name, added description too.

File: ui/admin/tests/acceptance/host-catalogs/hosts/update-test.js [hunk 3]
```diff
@@ -108,68 +125,82 @@ module('Acceptance | host-catalogs | hosts | update', function (hooks) {
           details: {
             request_fields: [
               {
-                name: 'name',
-                description: 'Name is required.',
+                name: 'address',
```
Comment (calcaide) [inline ui/admin/tests/acceptance/host-catalogs/hosts/update-test.js:128]: Changed the field we mark as invalid. `name` is an optional field within hosts, but address is required one, so this change is to make the test realistic.

File: ui/admin/tests/acceptance/host-catalogs/hosts/update-test.js [hunk 4]
```diff
@@ -108,68 +125,82 @@ module('Acceptance | host-catalogs | hosts | update', function (hooks) {
...
-    assert.ok(
-      find('[data-test-error-message-name]').textContent.trim(),
-      'Name is required.',
-    );
+    assert.dom(selectors.FIELD_ADDRESS_ERROR).hasText('Address is required.');
   });
 
   test('can discard unsaved host changes via dialog', async function (assert) {
-    assert.expect(5);
+    assert.expect(7);
```
Comment (calcaide) [inline ui/admin/tests/acceptance/host-catalogs/hosts/update-test.js:149]: Add more assertions since I also add not just 1 field change but 2.

File: ui/admin/tests/helpers/selectors.js [hunk 5]
```diff
@@ -31,3 +31,9 @@ export const ALERT_TOAST_BODY =
   '[data-test-toast-notification] .hds-alert__description';
 export const ALERT_TOAST_DISMISS =
   '[data-test-toast-notification] .hds-dismiss-button';
+
+export const DIALOG_UNSAVED_CHANGES = '.rose-dialog';
```
Comment (calcaide) [inline ui/admin/tests/helpers/selectors.js:35]: `.rose-dialog` being something quite generalistic, but for the usage of this selector should be fine. Open to add a `data-test-*` attribute if you think is a better solution or open to discuss other solutions.

Comment (cameronperera) [reply ui/admin/tests/helpers/selectors.js:35]: I think this works. The "quite generalistic" selectors like `.rose-dialog` are the whole reason we created these new selectors. When I read the new selectors name, I immediately knew what it was referring to so I am good with it.

**Review by cameronperera — APPROVED** [review]

**Review by ZedLi — APPROVED** [review]
Thanks for the update!
